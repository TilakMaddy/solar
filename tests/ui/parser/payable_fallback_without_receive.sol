contract P1 {
    fallback() external payable {}
}

contract P2 is P1 {
    receive() external payable {}
}

contract P3 {
    fallback() external payable {}

    receive() external payable {}
}

contract P4 is P1 {}
