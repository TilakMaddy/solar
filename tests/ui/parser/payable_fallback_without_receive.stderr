warning: contract has a payable fallback function, but no receive ether function
  --> ROOT/tests/ui/parser/payable_fallback_without_receive.sol:LL:CC
   |
LL | contract P1 {
   |          --
   |
   = help: consider adding a `receive() external payable { ... }`

warning: contract has a payable fallback function, but no receive ether function
  --> ROOT/tests/ui/parser/payable_fallback_without_receive.sol:LL:CC
   |
LL | contract P4 is P1 {}
   |          --
   |
   = help: consider adding a `receive() external payable { ... }`

warning: 2 warnings emitted

